### PR TITLE
Handling concurrency in the TodoTemplate project (#1796)

### LINKED ISSUE #1796: Concurrency handling for the TodoTemplate project
It is required to handle the concurrency issues in the template projects requests.

### PR DESCRIPTION
**Pull request by milad-karimifard**
this closes #1796

### CODE REVIEW

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs [hunk 1]
```diff
@@ -19,6 +19,59 @@ protected override void OnModelCreating(ModelBuilder builder)
...
+
+    public override int SaveChanges()
+    {
+        try
+        {
+            return base.SaveChanges();
+        }
+        catch (DbUpdateConcurrencyException exception)
+        {
+            throw new ConflictException("Database concurrency exception happens", exception);
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs:32]: Use `ErrorStrings` and `nameof` instead

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/TodoItemController.cs [hunk 2]
```diff
@@ -58,6 +59,7 @@ public async Task Put(TodoItemDto dto, CancellationToken cancellationToken)
 
         var updatedTodoItem = _mapper.Map(dto, todoItemToUpdate);
 
+        _dbContext.Entry(updatedTodoItem).Property(o => o.ConcurrencyStamp).OriginalValue = dto.ConcurrencyStamp!;
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/TodoItemController.cs:62]: Why do we need this?

**Review by ysmoradi — CHANGES_REQUESTED** [review]

**Comment by ysmoradi** [comment]
Let's only focus on mapping `DbConcurrencyException`, an Unknown exception to `ConflicException`, a known exception which has its own HTTP status code by overriding `OnSaveChanges` of ef core db context
What do you think? @msynk

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs (hunk 1 above)
Comment (ysmoradi) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs:32]: Could you please propose a meaningful error message here? @msynk

**Comment by ysmoradi** [comment]
@msynk PING!

**Comment by ysmoradi** [comment]
PING! @msynk

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs (hunk 1 above)
Comment (msynk) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs:32]: how about something like this:

`The record was modified by another user after you got the original data. the operation was canceled.`

Comment (ysmoradi) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs:32]: @milad-karimifard Please move this to ErrorStrings and use that via `nameof`

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/TodoItemController.cs [hunk 3]
```diff
@@ -40,7 +40,7 @@ public async Task<TodoItemDto> Get(int id, CancellationToken cancellationToken)
     public async Task Post(TodoItemDto dto, CancellationToken cancellationToken)
     {
         var todoItemToAdd = _mapper.Map<TodoItem>(dto);
-
+
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/TodoItemController.cs:43]: Revert all changes in this file

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs [hunk 4]
```diff
@@ -19,6 +19,34 @@ protected override void OnModelCreating(ModelBuilder builder)
         ConfigIdentityTables(builder);
     }
 
+    #region Override Save Changes
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Data/TodoTemplateDbContext.cs:22]: Remove this region

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Shared/Dtos/TodoItem/TodoItemDto.cs [hunk 5]
```diff
@@ -6,6 +6,5 @@ public class TodoItemDto
     public string? Title { get; set; }
     public DateTimeOffset Date { get; set; }
     public bool IsDone { get; set; }
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Shared/Dtos/TodoItem/TodoItemDto.cs:8]: Revert this file changes

**Review by ysmoradi — CHANGES_REQUESTED** [review]

**Review by ysmoradi — APPROVED** [review]

**Review by msynk — APPROVED** [review]

**Pull request merged**